ionet-setup.sh

```diff
@@ -32,6 +32,16 @@ else
 # Check if nvidia-smi is available and working
     if command -v nvidia-smi && nvidia-smi | grep CUDA | grep -vi 'n/a' &>/dev/null; then
         echo "CUDA drivers already installed as nvidia-smi works."
+
+        # Extract the CUDA version from nvidia-smi output and check if it meets the minimum requirement
+        cuda_version=$(nvidia-smi | grep "CUDA Version" | sed 's/.*CUDA Version: \([0-9]*\.[0-9]*\).*/\1/')
+        if [ "$(echo "$cuda_version >= 11.8" | bc -l)" -eq 1 ]; then
+            echo "CUDA version $cuda_version is installed and meets the minimum requirement of 11.8."
+        else
+            echo "CUDA version $cuda_version is installed but does not meet the minimum requirement of 11.8. Please upgrade CUDA."
+            exit 1
```

Comment:
I think this upgrade should be done by the script itself, as it's expected to be as unattended as possible.
Please update the installation logic below.